test/junit/src/test/scala/strawman/collection/mutable/HashMapTest.scala

```diff
@@ -35,4 +36,17 @@ class HashMapTest {
     hm.put(0, 0)
     hm.getOrElseUpdate(0, throw new AssertionError())
   }
+
+  @Test
+  def getOrElseUpdate_keyIdempotence(): Unit = {
+    val map = mutable.HashMap[String, String]()
+
+    val key = "key"
+    map.getOrElseUpdate(key, {
+      map.getOrElseUpdate(key, "value1")
+      "value2"
+    })
+
+    assertEquals(List((key, "value2")), List.from(map.iterator()))
```

Comment:
Maybe it’s simpler to write `map.toList` instead of `List.from(map.iterator())`?

Reply:
Fixed.
Something wrong with me. Thought `toList` is deprecated.